src/nvim/search.c

```diff
@@ -3878,24 +3881,34 @@ current_search (
 
 /*
  * Check if the pattern is one character or zero-width.
+ * If move is TRUE, check from the beginning of the buffer,
+ * else from the current cursor position.
  * Returns TRUE, FALSE or -1 for failure.
  */
-static int is_one_char(char_u *pattern)
+static int is_one_char(char_u *pattern, int move)
```

Comment:
You could change this to `bool` and use `true`/`false` instead of `TRUE`/`FALSE` in the invocations and the added comments.